src/pages/StocktakeManagePage.js

```diff
@@ -9,148 +9,138 @@ import React, { useCallback, useEffect } from 'react';
 import PropTypes from 'prop-types';
 import { View } from 'react-native';
 import { SearchBar } from 'react-native-ui-components';
-import { buttonStrings, modalStrings } from '../localization';
+
 import { UIDatabase } from '../database';
+
+import { usePageReducer } from '../hooks';
+import { recordKeyExtractor, getItemLayout } from './dataTableUtilities';
+import { createStocktake, updateStocktake } from '../navigation/actions';
+
 import { BottomTextEditor } from '../widgets/modals';
-import { ToggleBar } from '../widgets';
+import { ToggleBar, DataTablePageView } from '../widgets';
 import { DataTable, DataTableHeaderRow, DataTableRow } from '../widgets/DataTable';
-import {
-  sortData,
-  filterData,
-  selectRow,
-  deselectRow,
-  selectAll,
-  deselectAll,
-  hideStockOut,
-  showStockOut,
-  selectItems,
-  editName,
-} from './dataTableUtilities/actions';
-
-import globalStyles, { SUSSOL_ORANGE, newDataTableStyles, newPageStyles } from '../globalStyles';
-import usePageReducer from '../hooks/usePageReducer';
-import DataTablePageView from './containers/DataTablePageView';
-import { createStocktake, addItemsToStocktake } from '../navigation/actions';
-import { recordKeyExtractor, getItemLayout } from './dataTableUtilities/utilities';
 
-export const StocktakeManagePage = ({
-  routeName,
-  dispatch: reduxDispatch,
-  stocktake,
-  runWithLoadingIndicator,
-}) => {
-  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(routeName, {
-    pageObject: stocktake,
-    backingData: UIDatabase.objects('Item'),
-    data: UIDatabase.objects('Item')
-      .sorted('name')
-      .slice(),
+import { buttonStrings, modalStrings } from '../localization';
+import { SUSSOL_ORANGE, newPageStyles } from '../globalStyles';
+
+const stateInitialiser = pageObject => {
+  const backingData = UIDatabase.objects('Item');
+  return {
+    pageObject,
+    backingData,
+    data: backingData.sorted('name').slice(),
     keyExtractor: recordKeyExtractor,
     dataState: new Map(),
     searchTerm: '',
     filterDataKeys: ['name', 'code'],
+    name: pageObject ? pageObject.name : '',
     sortBy: 'name',
     isAscending: true,
     hasSelection: false,
     allSelected: false,
     showAll: true,
-    name: stocktake ? stocktake.name : '',
-  });
+  };
+};
+
+export const StocktakeManagePage = ({
+  routeName,
+  dispatch: reduxDispatch,
+  stocktake,
+  runWithLoadingIndicator,
+}) => {
+  const [state, dispatch, instantDebouncedDispatch, debouncedDispatch] = usePageReducer(
+    routeName,
+    {},
+    stateInitialiser,
+    stocktake
```

Comment:
I think `usePageReducer` could be changed:

`usePageReducer(routeName, initialValue, initialiser)` - such that if `initialiser` is passed, `initialValue` is the input - otherwise just use `initialValue` as the initial state.